### DISCUSSION TITLE
Issue with accessing authelia on my local network

### DISCUSSION THREAD

**Discussion by neonraver** [authelia/authelia#6905] [Q&A]
I have had authelia set up successful for a while, putting some of my server's services behind it when accessing via reverse proxy. Today everything stopped working - no idea why - but I made some copies, removed everything, and then redeployed my docker container. Authelia is now back up and running, and it is working externally which is what I need, but curiously I am unable to load it internally when going to my server's IP address and authelia port. I have this exact address setup on my reverse proxy and can access it via that, just not internally. When I do try and access it, I get a red error box at the top right saying "There was an issue fetching the current user state".
In the logs, this is the error I get:
`2024-03-15T22:12:28.350593512Z time="2024-03-15T22:12:28Z" level=error msg="Error occurred retrieving user session" error="unable to retrieve session cookie domain provider: no configured session cookie domain matches the url 'http://192.168.1.107:9091/api/state'" method=GET path=/api/state remote_ip=192.168.1.150 stack="github.com/authelia/authelia/v4/internal/handlers/handler_state.go:16  StateGET\ngithub.com/authelia/authelia/v4/internal/middlewares/bridge.go:54      handleRouter.(*BridgeBuilder).Build.func5.1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:56     SecurityHeadersCSPNone.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:82     SecurityHeadersNoStore.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:12     SecurityHeaders.func1\ngithub.com/fasthttp/router@v1.5.0/router.go:441                        (*Router).Handler\ngithub.com/authelia/authelia/v4/internal/middlewares/log_request.go:14 handleRouter.LogRequest.func40\ngithub.com/authelia/authelia/v4/internal/middlewares/errors.go:38      RecoverPanic.func1\ngithub.com/valyala/fasthttp@v1.52.0/server.go:2374                     (*Server).serveConn\ngithub.com/valyala/fasthttp@v1.52.0/workerpool.go:224                  (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.52.0/workerpool.go:196                  (*workerPool).getCh.func1\nruntime/asm_arm64.s:1222                                               goexit"`
I'm not particularly proficient at troubleshooting, so any help on why this is happening would be much appreciated!

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

The error message you're encountering, "unable to retrieve session cookie domain provider: no configured session cookie domain matches the url 'http://192.168.1.107:9091/api/state'", suggests that Authelia is unable to match the domain of the URL you're accessing (`192.168.1.107`) with any of the session cookie domains configured in Authelia's configuration.

This issue typically arises when Authelia's configuration does not explicitly include the domain (in this case, the IP address) you're trying to access it from. Authelia uses the session cookie domain configuration to determine if it should serve the request or not. If the domain of the request does not match any of the configured session cookie domains, Authelia will not be able to retrieve or set the necessary session cookies, leading to the error you're seeing.

To resolve this issue, you'll need to adjust your Authelia configuration to include the internal IP address (`192.168.1.107`) as a valid session cookie domain. This involves modifying the `session` configuration section in your `configuration.yml` file, specifically the `domain` attribute to include the IP address you're accessing Authelia from internally.

Here's a general approach to how you might adjust the configuration:

1. Locate your Authelia `configuration.yml` file.
2. Find the `session` section.
3. In the `domain` attribute, ensure that the internal IP address (`192.168.1.107`) is included. If you're using multiple domains, you might need to specify them as a list.

Please note, the exact configuration might vary based on your specific setup and version of Authelia. If you're unsure about making these changes, it's a good idea to consult the Authelia documentation or seek further assistance with the specific syntax and configuration options.

After making the necessary changes, restart your Authelia container for the changes to take effect. This should resolve the issue and allow you to access Authelia internally using the IP address.

If you continue to experience issues or if the configuration changes do not resolve the problem, it might be helpful to review the Authelia logs for any additional errors or warnings that could provide further insight into the issue.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
What do you mean internally? It seems like you are trying to access it via direct IP which is not really supported.

**Reply by DopeCoder01** [reply]
Have you found a solution? This is literally the only source I can find which has this issue. The AI above says to make a list, but a standard list is not accepted in the yaml so it appears only one entry can be set into domain. It's really bugging me as no tutorial whatsoever mentions these cookie options and explains it in relatively simple terms.

**Reply by neonraver** [reply]
> Have you found a solution? This is literally the only source I can find which has this issue. The AI above says to make a list, but a standard list is not accepted in the yaml so it appears only one entry can be set into domain. It's really bugging me as no tutorial whatsoever mentions these cookie options and explains it in relatively simple terms.

No, couldn't find a solution but it doesn't really bother me. I don't need it on my local network as it's just me, so I'm hustling living with it.

**Reply by DopeCoder01** [reply]
Ah, ok! Just realised indeed that it does not really matter much. Thought I needed to be able to do more configuring there but I saw that wrongly. Thanks for the reply though!

**Reply by hendrik1120** [reply]
The AI response is not necessarily accurate. You can use Authelia in a local network perfectly fine. You just need a local dns server, you could use pihole if you have that setup already. After that, it's just the normal authelia setup.

**Reply by DopeCoder01** [reply]
@hendrik1120 I didnt think of that! Setting up a local DNS server was the next step I planned after setting up a reverse proxy (and authelia). Thanks for the insight :)

**Reply by hendrik1120** [reply]
Happy to help. I can recommend this video/guide to get you started. https://technotim.live/posts/traefik-3-docker-certificates/
It will basically set you up with traefik, local dns and valid ssl certs. If you need any help, feel free to ask in the authelia discord support channel.

**Reply by DopeCoder01** [reply]
Thanks a lot! Will take a look into it. Althoug I use Nginx, I learned along the way that sometimes other manuals can really help understand what you are doing instead of blatantly following instructions. (Although it's mostly the latter for me at the moment :))

**Reply by electricmessiah** [reply]
This happened to me on a container reboot. I'm not sure the cause.

Error occurred retrieving user session" error="unable to retrieve session cookie domain provider: no configured session cookie domain matches the url 'http://192.168.1.10:9091/api/state'"

**Reply by Crowley723** [reply]
> This happened to me on a container reboot. I'm not sure the cause.
> 
> Error occurred retrieving user session" error="unable to retrieve session cookie domain provider: no configured session cookie domain matches the url 'http://192.168.1.10:9091/api/state'" 

This is likely because your attempting to access authelia directly via ip:port instead of through a reverse proxy. Authelia requires headers that the proxy adds to the requests.
